lua/lspconfig/psalm.lua

```diff
@@ -0,0 +1,28 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local server_name = 'psalm'
+local bin_name = 'bin/psalm-language-server'
+
+configs[server_name] = {
+  default_config = {
+    cmd = { bin_name },
+    filetypes = { 'php' },
+    root_dir = function(pattern)
+      local cwd = vim.loop.cwd()
+      local root = util.root_pattern('psalm.xml', 'composer.json', '.git')(pattern)
```

Comment:
There's now another open pull request for Psalm support (GH-1260) that mentions `psalm.xml.dist`.